llvm/lib/IR/RepoTicket.cpp

```diff
@@ -77,76 +78,55 @@ const Constant *getAliasee(const GlobalAlias *GA) {
-  }
-
-  return {updateInitialDigestAndGetDependencies(GO, GOHash, GOIMap, GONum),
-          true};
+static const GODigestState updateDigestAndGetGOState(const GlobalObject *GO,
+                                                     MD5 &GOHash,
+                                                     GOInfoMap &GOIMap) {
+  const GOInfo &GOInformation = GOIMap[GO];
+  GOHash.update(GOInformation.InitialDigest.Bytes);
+  return std::forward_as_tuple(GOInformation.Contributions,
+                               GOInformation.Dependencies, true);
 }
 
-GODigestState updateGODigestAndGetDependencies(const GlobalObject *GO,
-                                               MD5 &GOHash, GOInfoMap &GOIMap) {
-  if (auto GOMD = GO->getMetadata(LLVMContext::MD_repo_ticket)) {
-    if (const TicketNode *TN = dyn_cast<TicketNode>(GOMD)) {
+static GODigestState updateGODigestAndGetGOState(const GlobalObject *GO,
+                                                 MD5 &GOHash,
+                                                 GOInfoMap &GOIMap) {
+  if (const auto *const GOMD = GO->getMetadata(LLVMContext::MD_repo_ticket)) {
+    if (const TicketNode *const TN = dyn_cast<TicketNode>(GOMD)) {
       DigestType D = TN->getDigest();
       GOHash.update(D.Bytes);
-      return {GOIMap
-                  .try_emplace(GO, std::move(D), DependenciesType(),
-                               ContributionsType())
-                  .first->second.Dependencies,
-              true};
+      const GOInfo &GOInformation =
+          GOIMap.try_emplace(GO, std::move(D), SmallGOVec(), SmallGOVec())
+              .first->second;
+      return std::forward_as_tuple(GOInformation.Contributions,
+                                   GOInformation.Dependencies, true);
     }
     report_fatal_error("Failed to get TicketNode metadata!");
   }
-  GONumber GONum;
-  return {updateInitialDigestAndGetDependencies(GO, GOHash, GOIMap, GONum),
-          true};
-}
-
-GODigestState updateDigestUseContributions(const GlobalObject *GO, MD5 &GOHash,
-                                           GOInfoMap &GOIMap,
-                                           GVInfoMap GVIMap) {
-  bool Changed = false;
-  if (const auto &GV = dyn_cast<GlobalVariable>(GO)) {
-    if (GVIMap.count(GV)) {
-      GOHash.update(getTicket(GO)->getDigest().Bytes);
-      Changed = true;
-    }
-  }
-  return {GOIMap[GO].Dependencies, Changed};
+  return updateInitialDigestAndGetGOStatus(GO, GOHash, GOIMap);
 }
 
 // Update the GO's hash value by adding the hash of its dependents.
 template <typename Function>
-static bool
-updateDigestUseCallDependencies(const GlobalObject *GO, MD5 &GOHash,
-                                GOStateMap &Visited, GOInfoMap &GOIMap,
-                                Function UpdateDigestAndGetDependencies) {
+static bool updateDigestUseDependenciesAndContributions(
+    const GlobalObject *GO, MD5 &GOHash, GOStateMap &Visited, GOInfoMap &GOIMap,
+    Function UpdateDigestAndGetGOInfo) {
```

Comment:
Ah, the "And" functions are used to wrap something up as a Function argument. Maybe that's alright.

Reply:
Yes, you are right.